packages/flutter/lib/src/widgets/restoration_properties.dart

```diff
@@ -306,11 +306,11 @@ class RestorableBoolN extends _RestorablePrimitiveValueN<bool?> {
 /// See also:
 ///
 ///  * [RestorableNum] for the non-nullable version of this class.
-class RestorableNumN<T extends num?> extends _RestorablePrimitiveValueN<num?> {
+class RestorableNumN<T extends num?> extends _RestorablePrimitiveValueN<T> {
   /// Creates a [RestorableNumN].
   ///
   /// {@macro flutter.widgets.RestorableNum.constructor}
-  RestorableNumN(num? defaultValue) : super(defaultValue);
+  RestorableNumN(T defaultValue) : super(defaultValue);
```

Comment:
So this `T` should be only a num, int, or double right?  It's not possible to check that with an assertion is it?  I know Dart doesn't have union types.

Reply:
Yeah, I'm not sure if we can check it with an assertion. The way I've been enforcing the types is through the definition of T in the class, like `RestorableNumN<T extends num?>`

Reply:
Plus, I don't think we'll have to worry about T being something other than num, int or double. According the num documentation:
 * It is a compile-time error for any type other than [int] or [double] to attempt to extend or implement num.